docs/articles/concepts/joins.md

```diff
@@ -1,3 +1,775 @@
+
+Then we see the first name value. Since the table was updated in any
+way, it emits the current values for that key and keeps them in the
+table! If it doesn't know a value yet, it fills it in as `None`.
+
+| Key | Name Value | Email Value |
+| --- | ---------- | ----------- |
+| 123 | `"Bee"` | |
+
+```
+('123', ({'user_id': 123, 'name': 'Bee'}, None))
+```
+
+Then we see the second name value. The same thing happens
+
+| Key | Name Value | Email Value |
+| --- | ---------- | ----------- |
+| 123 | `"Bee"` | |
+| 456 | `"Hive"` | |
+
+```
+('456', ({'user_id': 456, 'name': 'Hive'}, None))
+```
+
+Now we see the first email value. The same rules apply, but now since
+there are values for all the sides, we see them all in the output. The
+state for that key is not cleared!
+
+| Key | Name Value | Email Value |
+| --- | ---------- | ----------- |
+| 123 | `"Bee"` | `"bee@bytewax.io"` |
+| 456 | `"Hive"` | |
+
+```
+('123', ('Bee', 'bee@bytewax.io'))
+```
+
+Then we see the second email value.
+
+| Key | Name Value | Email Value |
+| --- | ---------- | ----------- |
+| 123 | `"Bee"` | `"bee@bytewax.io"` |
+| 456 | `"Hive"` | `"hive@bytewax.io"` |
+
+```
+('456', ('Hive', 'hive@bytewax.io'))
+```
+
+Note that none of the state we have seen has been cleared. This now
+means when we see the updated Bee email, we'll see some output!
+
+| Key | Name Value | Email Value |
+| --- | ---------- | ----------- |
+| 123 | `"Bee"` | `"queen@bytewax.io"` |
+| 456 | `"Hive"` | `"hive@bytewax.io"` |
+
+```
+('123', ('Bee', 'queen@bytewax.io'))
+```
+
+So the running join is cool in that you can track updates to changes
+in values over time. _But this comes with a downside!_ Because we
+never throw away the state for a key, this state keeps growing in
+memory _forever_ if you keep adding keys. This might be the behavior
+you want, but realize that it does not come for free. A non-running
+streaming join is better if you know you'll only get one value for
+each side for each key, since it knows it can discard the state after
+sending the values downstream.
+
+## Windowed Joins
+
+The streaming join assumes that a key could come anywhere in the
+entire lifetime of a stream. This means it could possibly wait forever
+for a value that will never arrive. Another option is
+to use a **windowed join** that always flushes out the values for a
+key whenever the time-based window closes. You can use this if you
+need to know the join values over an infinite stream when you aren't
+sure that you'll see values on all sides of the join.
+
+Bytewax provides the operators
+[`join_window`](/apidocs/bytewax.operators/window/index#bytewax.operators.window.join_window),
+and
+[`join_window_named`](/apidocs/bytewax.operators/window/index#bytewax.operators.window.join_window)
+to implement this.
+
+For the details of all the types of windows you can define and
+explanation of the parameters, see our [windowing
+documentation](windowing.md). I'm going to use a simple 1 hour
```

Comment:
It's better to stick to the same-person style for consistency and in this sort of docs it's better to use "**We** are going to use" instead of "**I'm** going to use", since it's technically not a blog post